### PR TITLE
[GPU] Fix handle reshape not to be applied for type conversion reorder user

### PR DESCRIPTION
**Pull request by kelvinchoi-intel**
### Details:
 - *In case reshape has an reorder user which truncation mode is true, the reorder should not be splited.*

### Tickets:
 - *110907*

### CODE REVIEW

File: report_op.xml [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+		<Tan-1 opsets="1 10 11 12 2 3 4 5 6 7 8 9" />
+		<Tanh-1 opsets="1 10 11 12 2 3 4 5 6 7 8 9" />
+		<TensorIterator-1 opsets="1 10 11 12 2 3 4 5 6 7 8 9" />
+		<Tile-1 opsets="1 10 11 12 2 3 4 5 6 7 8 9" />
+		<TopK-1 opsets="1 2" />
+		<TopK-11 opsets="11 12" />
+		<TopK-3 opsets="10 3 4 5 6 7 8 9" />
+		<Transpose-1 opsets="1 10 11 12 2 3 4 5 6 7 8 9" />
+		<Unique-10 opsets="10 11 12" />
+		<Unsqueeze-1 opsets="1 10 11 12 2 3 4 5 6 7 8 9" />
```
Comment (vladimir-paramuzov) [inline report_op.xml:217]: remove these report files

File: src/plugins/intel_gpu/src/graph/graph_optimizer/handle_reshape.cpp [hunk 2]
```diff
@@ -95,10 +96,20 @@ void handle_reshape::run(program& p) {
             std::vector<program_node*> reorder_node_to_split;
             std::vector<program_node*> onednn_users;
 
+            // In case reshape is used as an user of loop operation,
```
Comment (vladimir-paramuzov) [inline src/plugins/intel_gpu/src/graph/graph_optimizer/handle_reshape.cpp:99]: Need a test for this change

**Review by vladimir-paramuzov — CHANGES_REQUESTED** [review]

File: src/plugins/intel_gpu/src/graph/graph_optimizer/handle_reshape.cpp [hunk 3]
```diff
@@ -88,17 +89,41 @@ void handle_reshape::run(program& p) {
...
-            node->get_output_layout();
+            auto output_layout = node->get_output_layout();
 
             // vector for storing nodes that are reorder type, for which splitted primitives are needed (except for the
             // first one where orginal reshape will be used)
             std::vector<program_node*> reorder_node_to_split;
             std::vector<program_node*> onednn_users;
 
+            // In case reshape is used as an user of loop operation and the type is integer,
+            // the output of reshape node needs not to change from ngraph.
```
Comment (vurusovs) [inline src/plugins/intel_gpu/src/graph/graph_optimizer/handle_reshape.cpp:100]: Please, remove `ngraph`

File: report_op.xml (hunk 1 above)
Comment (kelvinchoi-intel) [reply report_op.xml]: done

File: src/plugins/intel_gpu/src/graph/graph_optimizer/handle_reshape.cpp [hunk 4]
```diff
@@ -88,17 +89,41 @@ void handle_reshape::run(program& p) {
...
+                            loop_prim.get_initial_execution_id().compare(node_id) == 0 ||
+                            loop_prim.get_current_iteration_id().compare(node_id) == 0 ||
+                            loop_prim.get_execution_condition_id().compare(node_id) == 0 ||
+                            loop_prim.get_num_iterations_id().compare(node_id) == 0) {
+                            need_reorder_node_to_split = false;
+                            break;
+                        }
+                    }
+                }
+            }
```
Comment (isanghao) [inline src/plugins/intel_gpu/src/graph/graph_optimizer/handle_reshape.cpp:121]: Based on your explanation from the meeting, I think this is not about loop, but general issue. Could you fix the real problem, not the symptom-on-loop?

**Review by isanghao — CHANGES_REQUESTED** [review]

**Comment by yeonbok** [comment]
Please add a unittest.

File: src/plugins/intel_gpu/src/graph/graph_optimizer/handle_reshape.cpp (hunk 4 above)
Comment (kelvinchoi-intel) [reply src/plugins/intel_gpu/src/graph/graph_optimizer/handle_reshape.cpp]: Yes, it's updated with general fix for reshape and reorder.

File: src/plugins/intel_gpu/src/graph/graph_optimizer/handle_reshape.cpp (hunk 2 above)
Comment (kelvinchoi-intel) [reply src/plugins/intel_gpu/src/graph/graph_optimizer/handle_reshape.cpp]: Added an unittests

**Review by ahnyoung-paul — APPROVED** [review]

**Review by isanghao — APPROVED** [review]

**Pull request merged**